setup.py

```diff
@@ -71,6 +71,53 @@
 import distutils
 from setuptools import setup
 
+import distutils.ccompiler
+
+
+filename = 'simd_blitters_avx2'
+
+compiler_options = {
+    'unix': ('-mavx2',),
+    'msvc': ('/arch:AVX2',)
+}
+
+
+def spawn(self, cmd, **kwargs):
+    if (('ARM' not in platform.uname().version) and
+            platform.machine() != 'aarch64' and
+            'arm' not in platform.machine()):
```

Comment:
I'm not sure how this would affect our mac M1 builds. Currently, we have a bit of a hacky solution where our M1 builds are made on an intel mac, but cross compiled to `arm`. So in `setup.py`, `platform.machine()` and `platform.uname()` would still return intel info on our arm builds.

Since sse2 and avx are intel/AMD specific stuff, this check needs to check for that somehow, because checking for ARM to not match misses other exotic platforms like `ppc` and `s390x`

Reply:
I think I figured out a reasonable way to check for M1 mac cross compiling by using the fact that the arm-simd flags are set in the mac .sh file when building for M1 macs. 

There still seems to be ongoing debate over at CPython HQ on a standardised way to communicate a cross compilation scenario. At least I found an unresolved discussion & PEP from 2021:
https://github.com/benfogle/cross-compile-pep-draft/pull/1/files


Do you know of any list of supported pygame platforms? I could exclude them by `platform.machine()`/uname name as I've done for arm and perhaps provide a fall back compiler flag `-disable-avx2` to catch anything else we miss for really obscure platforms. Or I could perhaps reverse the check and just add a list of supported architectures (windows, non-m1 macs, non arm linux), or add an `-enable-avx2` flag that would could put on for our avx2 supporting wheels.

Reply:
hmmm I still don't think checking for `-enable-arm-(neon|simd)` would be accurate on our arm mac builds, because these flags are not passed to `setup.py` itself. One way I can think of check of is checking for `MAC_ARCH` environment variable, which is a custom thing defined by our mac CI, that would be `arm64` on M1 and `x86_64` otherwise

> Do you know of any list of supported pygame platforms?

AFAIK there is no such list for pygame, but I find [this wiki](https://en.wikipedia.org/wiki/Uname) handy that keeps a list of possible `uname` return codes.

I think the test can look something like this, which tests for intel/amd specifically, but allows both 32-bit and 64-bit variants
```py
mach = platform.machine()
if (
    mach.startswith("x86") or mach.startswith("i686") or mach.lower() == "amd64"
) and os.environ.get("MAC_ARCH") != "arm64":
    # do intel specific stuff here
    pass
```